### PR TITLE
Fixes global server bans and adds new config options

### LINKED ISSUE #806: We need 2 new config options for, uh, reasons
@ike709 

1: We need a config option to disable the income of beecoins
2: We need a config option to make a server not care about global bans from other servers, and only the local/global bans on itself.

### PR DESCRIPTION
**Pull request by ike709**
Closes #806 

Also fixes global server bans. `IsBanned()` uses `is_banned_with_details()` to check if a player is banned from the server. @MarkSuckerberg only added a global vs local ban check to `is_banned()`, which is only used to check if someone is jobbanned.

## Changelog
:cl:
fix: Players locally banned from one of the servers are no longer globally banned from both servers. This fix applies to all bans retroactively. We apologize for the inconvenience.
config: Added some new config options.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
DNM until all of the following people sign off on this:
@qwertyquerty 
@Crossedfall 
@MarkSuckerberg

**Review by MarkSuckerberg — APPROVED** [review]

File: code/controllers/configuration/entries/general.dm [hunk 1]
```diff
@@ -494,3 +494,7 @@
 
 /datum/config_entry/string/metacurrency_name
 	config_entry_value = "MetaCoin"
+
+/datum/config_entry/flag/grant_metacurency
```
Comment (qwertyquerty) [inline code/controllers/configuration/entries/general.dm]: You spelled currency wrong

**Review by qwertyquerty — APPROVED** [review]

File: code/controllers/configuration/entries/general.dm (hunk 1 above)
Comment (ike709) [reply code/controllers/configuration/entries/general.dm]:
```suggestion
/datum/config_entry/flag/grant_metacurrency
```

File: code/modules/metacoin/metacoin.dm [hunk 2]
```diff
@@ -38,6 +38,8 @@
 		to_chat(src, "<span class='rose bold'>Your new metacoin balance is [mc_count]!</span>")
 
 /client/proc/inc_metabalance(mc_count, ann=TRUE, reason=null)
+	if(mc_count >= 0 && !CONFIG_GET(flag/grant_metacurency))
```
Comment (ike709) [inline code/modules/metacoin/metacoin.dm]:
```suggestion
	if(mc_count >= 0 && !CONFIG_GET(flag/grant_metacurrency))
```

**Comment by Crossedfall** [comment]
Config options applied. Test merged onto Both servers.

**Review by Crossedfall — APPROVED** [review]
Worked within my tests. Everything else looks good!

**Pull request merged**